keel-core/src/main/kotlin/com/netflix/spinnaker/keel/services/ApplicationService.kt

```diff
@@ -245,12 +253,28 @@ class ApplicationService(
           potentialSummary
         }
       }
-      else -> repository.getArtifactSummaryInEnvironment(
-        deliveryConfig = deliveryConfig,
-        environmentName = environmentName,
-        artifactReference = artifact.reference,
-        version = version
-      )
+      DEPLOYING -> {
+        val olderGitMetadata = repository.getGitMetadataByPromotionStatus(deliveryConfig, environmentName, artifact.reference, CURRENT.name)
+        potentialSummary?.copy(
+          // comparing DEPLOYING (version is question, new code) vs. CURRENT (old code)
+          diffLink = generateDiffLink(artifactGitMetadata, olderGitMetadata)
+        )
+      }
+      PREVIOUS -> {
+        val newerGitMetadata = potentialSummary?.replacedBy?.let { getArtifactInstance(artifact, it)?.gitMetadata }
+        potentialSummary?.copy(
+          //comparing PREVIOUS (version is question, old code) vs. the version which replaced it (new code)
+          diffLink = generateDiffLink(newerGitMetadata, artifactGitMetadata)
+        )
+      }
+      CURRENT -> {
+        val olderGitMetadata = repository.getGitMetadataByPromotionStatus(deliveryConfig, environmentName, artifact.reference, PREVIOUS.name)
```

Comment:
My understanding is that there may be potentially many versions with a `PREVIOUS` status. Which one will this pick?

Reply:
the function `getGitMetadataByPromotionStatus` is querying `environemnt_artifact_version` by promotion status and other params, and will return the last `deployed_at` version. Using this version, we are querying to get its `gitMetadata` for the comparsion. Do you think it could be error-prone?

Reply:
Sorting by `deployed_at` and getting the latest sounds OK to me.